### PR TITLE
Reset all locale environment variables before running svn commands

### LINKED ISSUE #3255: source_control/subversion.py needs to reset LC_MESSAGES
##### Issue Type:

- Bug Report
##### Plugin Name:

subversion
##### Ansible Version:

```
ansible-playbook 2.1.0 (devel 86080fbaa9) last updated 2016/03/15 19:12:05 (GMT +200)
  lib/ansible/modules/core: (detached HEAD a8841e6834) last updated 2016/03/15 19:40:46 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 45bba8ec64) last updated 2016/03/15 19:40:46 (GMT +200)
  config file = /var/tmp/subversion.bug/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

Default
##### Environment:

Debian GNU Linux 8.3
##### Summary:

Subversion module needs to reset LC_MESSAGES, not just LANG, since it screenscrapes the output of the svn(1) client. #2462 was the same issue for git.
##### Steps To Reproduce:

```
- hosts: all
  tasks:
    - shell: rm -fr {{ playbook_dir }}/repo {{ playbook_dir }}/working && svnadmin create {{ playbook_dir }}/repo
    - subversion: repo=file://{{ playbook_dir }}/repo dest={{ playbook_dir }}/working
EOF
```

##### Expected Results:

Success
##### Actual Results:

```
$ LANG=fr_FR.utf8 ansible-playbook -i localhost, svn_co.yml 

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [command] *****************************************************************
changed: [localhost]
 [WARNING]: Consider using file module with state=absent rather than running rm


TASK [subversion] **************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/quinot/.ansible/tmp/ansible-tmp-1458067626.47-249928368830364/subversion\", line 2323, in <module>\r\n    main()\r\n  File \"/home/quinot/.ansible/tmp/ansible-tmp-1458067626.47-249928368830364/subversion\", line 259, in main\r\n    after = svn.get_revision()\r\n  File \"/home/quinot/.ansible/tmp/ansible-tmp-1458067626.47-249928368830364/subversion\", line 167, in get_revision\r\n    rev = re.search(r'^Revision:.*$', text, re.MULTILINE).group(0)\r\nAttributeError: 'NoneType' object has no attribute 'group'\r\n", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @svn_co.retry

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=1
```

### PR DESCRIPTION
**Pull request by quinot**
##### Issue Type:
- Bugfix Pull Request
##### Plugin Name:

subversion
##### Summary:

Reset all locale environment variables so that svn(1) output can be parsed.

Fixes #3255

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @quinot. To the current maintainers, @dsummersl please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: source_control/subversion.py [hunk 1]
```diff
@@ -216,7 +216,10 @@ def main():
     export = module.params['export']
     switch = module.params['switch']
 
-    os.environ['LANG'] = 'C'
+    # We screenscrape a huge amount of git commands so use C locale anytime we
```
Comment (mscherer) [inline source_control/subversion.py]: The comment is wrong, should be 'svn', not 'git'

Comment (quinot) [inline source_control/subversion.py]: Good catch, copy/paste error. Pull request updated.

**Comment by abadger** [comment]
Looks good to me.  Thanks @quinot!

**Pull request merged**